x-pack/solutions/security/plugins/security_solution/server/lib/siem_migrations/dashboards/data/dashboard_migrations_dashboards_client.ts

```diff
@@ -59,6 +64,61 @@ export class DashboardMigrationsDataDashboardsClient extends SiemMigrationsDataB
     }
   }
 
+  /** Returns functions to iterate over all the search results in batches */
+  async searchBatches(migrationId: string, filters: SearchFilters = {}) {
+    const index = await this.getIndexName();
+    const query: QueryDslQueryContainer = {
+      bool: {
+        filter: [
+          { term: { migration_id: migrationId } },
+          ...(filters.ids ? [{ terms: { _id: filters.ids } }] : []),
+          ...(filters.installable ? [{ term: { status: SiemMigrationStatus.COMPLETED } }] : []),
+        ],
+      },
+    };
+
+    const search = {
+      index,
+      query,
+      sort: [{ '@timestamp': 'asc' }],
+      size: 100,
+    };
+
+    return this.getSearchBatches<DashboardMigrationDashboard>(search);
+  }
+
+  /** Updates dashboard migration documents */
+  async update(updates: Array<{ id: string; elastic_dashboard: { id: string } }>): Promise<void> {
```

Comment:
It would be great to use the proper type for this parameter. We'd need to create `UpdateDashboardMigrationDashboard`, mirroring the `UpdateRuleMigrationRule`.